src/Randomizer.SMZ3/FileData/PatcherServiceData.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Randomizer.Data.Configuration;
+using Randomizer.Data.Configuration.ConfigFiles;
+using Randomizer.Data.Configuration.ConfigTypes;
+using Randomizer.Data.Options;
+using Randomizer.Data.WorldData;
+using Randomizer.Data.WorldData.Regions;
+
+namespace Randomizer.SMZ3.FileData;
+
+/// <summary>
+/// Class used to house data utilized by the PatcherService
+/// </summary>
+public class PatcherServiceData
```

Comment:
This class feels kinda... dirty to me. I'm not sure what a better name would be, though.

Reply:
Yeah, I'm not sure what a better name for this would be. The only other idea I kind of mulled over was to have this be a "GetPatchesRequest" class instead and then the calling functions would instantiate that request object and pass it into the PatcherService.GetPatches method.

My main hesitation there was needing the metadata configs and passing global things like that into a request object feels a bit gross to me. I thought one work around for that could be to have the patches be dynamically added via dependency injection like the tracker modules are and have the configs dynamically injected when needed, but I was a bit concerned about that due to the lack of control over the order potentially causing problems when people add new patches.

So in the end, the best idea I've got would be to change this to a "GetPatchesRequest" which is built by the classes that call the PatcherService.GetPatches, have the PatcherService populate the metadata configs, then start passing that along to all the rom patches. Still doesn't feel _great_ in the end, but it's probably closer to normal C# paradigms than the current setup.

Reply:
You could still go the dependency injection route and simply add a `int Order` property to `RomPatch` to control the order in which they are applied. Maybe have it default to 0 and then you can use negative and positive values to make it apply later or earlier?

Reply:
True. I can look into that. Admittedly I could see being able to inject anything could be useful.